nucypher/blockchain/eth/clients.py

```diff
@@ -224,6 +228,12 @@ def sign_message(self, account: str, message: bytes) -> str:
 
 class GethClient(Web3Client):
 
+    @classmethod
+    def _get_variant(cls, w3):
+        if 'infura' in w3.provider.endpoint_uri:
```

Comment:
@vepkenez - Some web3 provider instances do not have the `endpoint_uri` attribute

Reply:
you could also do 'if `infura' in getattr(w3.provider, 'endpoint_uri', ''):`
either